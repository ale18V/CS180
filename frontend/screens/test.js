import React from 'react';
import { render, fireEvent} from '@testing-library/react-native';
import Landing from './landing';
import StatsCalendar from "../components/statsCalendar.jsx";
import StatsGraphs from "../components/statsGraphs.jsx";
import WorkoutInput from '../components/workoutInput.jsx';
import WorkoutSelect from '../components/workoutSelect.jsx';
import MarkSets from '../components/markSets.jsx';
import LogTab from "./logTab.jsx";



describe('Landing Page', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Landing />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders "Get Started" button and handles press event', () => {
    const navigation = { navigate: jest.fn() };
    const { getByText } = render(<Landing navigation={navigation} />);
    const getStartedButton = getByText('Get Started');
    fireEvent.press(getStartedButton);
    expect(navigation.navigate).toHaveBeenCalledWith('SignUp');
  });

});

describe('WorkoutInput component', () => {
  test('renders correctly', () => {
    const { getByText, getByPlaceholderText } = render(<WorkoutInput />);
    
    // Assert that required elements are rendered
    expect(getByText(' Either no plan, routine, or exercise found! Please add missing category before trying to log workout. If you are receiving this message incorrectly, click refresh data to update.')).toBeTruthy();
  });

  /* test('submits workout data on button click', () => {
    const { getByText, getByPlaceholderText } = render(<WorkoutInput />);
    
    // Mock fetch function
    global.fetch = jest.fn(() => Promise.resolve({ ok: true, json: () => Promise.resolve({ id: 1 }) }));

    // Enter data into input fields
    fireEvent.changeText(getByPlaceholderText('Weight'), '100');
    fireEvent.changeText(getByPlaceholderText('Reps'), '10');
    fireEvent.changeText(getByPlaceholderText('Sets'), '3');
    fireEvent.changeText(getByPlaceholderText('Rest'), '60');
    // Optionally, you can interact with the date picker if needed

    // Click the submit button
    fireEvent.press(getByText('Submit'));

    // Assert that fetch was called with the correct data
    expect(fetch).toHaveBeenCalledWith('http://localhost:8000/api/v1/workouts/', {
      method: 'POST',
      headers: {
        Authorization: 'Bearer your_access_token_here', // Add your access token here
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        date: expect.any(String), // You may need to adjust this depending on your test setup
        routine_id: -1, // Assuming routine_id starts as -1
      }),
    });
  }); */
});

describe('MarkSets component', () => {
  test('renders correctly', () => {
    const { getByText, getByPlaceholderText } = render(<MarkSets />);
    
    // Assert that required elements are rendered
    expect(getByText('Workout Notes')).toBeTruthy();
  });


});

describe("LogTab Component", () => {
  /* it("renders 'Add Workout' button", () => {
    const { getByText } = render(<LogTab />);
    const addWorkoutButton = getByText("Add a workout");
    expect(addWorkoutButton).toBeTruthy();
  });

  it("renders 'Done' button after pressing 'Add Workout' button", () => {
      const { getByText, queryByText } = render(<LogTab />);
      // 'Done' button should not be visible
      expect(queryByText("Done")).toBeNull();
      // Find and press the 'Add Workout' button
      const addWorkoutButton = getByText("Add a workout");
      fireEvent.press(addWorkoutButton);
      // After pressing the button, the 'Done' button should appear
      expect(getByText("Done")).toBeTruthy();
    }); */

    it('renders correctly', () => {
      const { toJSON } = render(<LogTab />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders buttons correctly', () => {
      const { getByText, queryByText } = render(<LogTab />);
      expect(getByText('My Workouts')).toBeTruthy();
      expect(getByText('Log Workouts')).toBeTruthy();
      expect(queryByText('Close View')).toBeNull(); //neither modal should be open
    });
  
    it('opens markModal when My Workouts button is pressed', () => {
      const { queryByText } = render(<LogTab />);
      fireEvent.press(queryByText('My Workouts'));
      expect(queryByText('Close View')).toBeTruthy(); // Expecting the notes modal to open
    });
  
    it('opens logModal when Log Workouts button is pressed', () => {
      const { queryByText } = render(<LogTab />);
      fireEvent.press(queryByText('Log Workouts'));
      expect(queryByText('Close View')).toBeTruthy(); // Expecting the log modal to open
    });

});


/* jest.mock('react-native-calendars', () => ({
    Agenda: jest.fn(() => null), 
  }));
  
  describe('Stats component', () => {
    test('renders correctly', () => {
      render(<Stats />);
    });
  }); */

  