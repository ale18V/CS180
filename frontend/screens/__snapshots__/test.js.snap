// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Page renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#e4edea",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f3fff5",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "white",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "borderWidth": 1.5,
              "flex": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#58a1a3",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
                "marginTop": 2,
              }
            }
          >
            My Workouts
          </Text>
          <Image
            source={1}
            style={
              {
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f3fff5",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "white",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "borderWidth": 1.5,
              "flex": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#58a1a3",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
                "marginBottom": 4.5,
              }
            }
          >
            Log Workouts
          </Text>
          <Image
            resizeMode="contain"
            source={1}
            style={
              {
                "height": 44,
                "width": 44,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`CreateExercise Component renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
            {
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
            },
          ]
        }
      >
        Create Exercise
      </Text>
      <
        color="black"
        name="keyboard-arrow-up"
        size={24}
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`CreateRoutine Component renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
            {
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
            },
          ]
        }
      >
        Create Routine
      </Text>
      <
        color="black"
        name="keyboard-arrow-up"
        size={24}
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`Landing Page renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        1077453733,
        3212897689,
      ]
    }
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        [
          {
            "marginTop": 80,
          },
          {
            "shadowColor": "rgba(0, 0, 0, 0.1)",
            "shadowOffset": {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 10,
          },
          {
            "height": 150,
            "width": 150,
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "fontSize": 24,
            "lineHeight": 32,
          },
          {
            "color": "#0f766e",
          },
          {
            "fontWeight": "700",
          },
          {
            "marginTop": 20,
          },
        ]
      }
    >
      Fitness Tracker
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#0d9488",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginTop": 160,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
          ]
        }
      >
        Get Started
      </Text>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;

exports[`LogTab Page renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#e4edea",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f3fff5",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "white",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "borderWidth": 1.5,
              "flex": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#58a1a3",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
                "marginTop": 2,
              }
            }
          >
            My Workouts
          </Text>
          <Image
            source={1}
            style={
              {
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#f3fff5",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "white",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "borderWidth": 1.5,
              "flex": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#58a1a3",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
                "marginBottom": 4.5,
              }
            }
          >
            Log Workouts
          </Text>
          <Image
            resizeMode="contain"
            source={1}
            style={
              {
                "height": 44,
                "width": 44,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`WorkoutSelect Component renders correctly 1`] = `
<View>
  <Text>
    Either no plan, routine, or exercise found! Please add missing category before trying to log workout. If you are receiving this message incorrectly, click refresh data to update. 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Refresh data
      </Text>
    </View>
  </View>
</View>
`;
